src/FffGcodeWriter.cpp

```diff
@@ -1612,7 +1625,8 @@ void FffGcodeWriter::processSkinPrintFeature(const SliceDataStorage& storage, La
 
     constexpr int extra_infill_shift = 0;
     constexpr coord_t offset_from_inner_skin_infill = 0;
-    Infill infill_comp(pattern, area, offset_from_inner_skin_infill, config.getLineWidth(), config.getLineWidth(), skin_overlap, skin_angle, gcode_layer.z, extra_infill_shift, perimeter_gaps_output);
+    const bool zig_zaggify_infill = pattern == EFillMethod::ZIG_ZAG;
```

Comment:
As this is sort of a challenging opinion, I discussed this change with Nallath and he disagreed, because the infill selection right now is very simple, and splitting a common case like zigzag into two settings makes zigzag hard to find/use. There's something to be said for that as well. So I'm not going to make this change.

Reply:
I thought this would be a good idea when actually implementing connect infill lines for all linear infill types. CURA-4017